services/util/client.test.js

```diff
@@ -83,7 +131,49 @@ describe('Client', () => {
       });
 
       expect(axiosClient.get).toHaveBeenCalledWith(apiUrl, { params });
-      expect(subject).toEqual({ foo: 'bar' });
+      expect(subject).toEqual({ pages });
+    });
+
+    it('aggregates the results of paged responses', async () => {
+      const titleString = 'Def_Leppard';
+      const titles = [titleString];
+      const prop = 'image';
+      const params = { ...defaultParams, prop, titles: titleString };
+      const continueParams = { tlcontinue: '46565|10|Tl', continue: '||' };
+
+      const mockedResponse1 = { data: { continue: continueParams, query: { pages: page1 } } };
+      const mockedResponse2 = { data: { batchcomplete: '', query: { pages: page2 } } };
+      axiosClient.get.mockResolvedValueOnce(mockedResponse1).mockResolvedValueOnce(mockedResponse2);
+
+      const client = new Client();
+      const subject = await client.getResultsFromApi(titles, 'image', wikiUrl);
+
+      expect(axiosClient.get).toHaveBeenCalledWith(apiUrl, { params });
+      expect(axiosClient.get).toHaveBeenCalledWith(apiUrl, {
+        params: { ...params, ...continueParams },
+      });
+      expect(subject).toEqual({ pages: { ...page1, ...page2 } });
+    });
+
+    it('correctly merges the results of paged responses if they are the same', async () => {
+      const titleString = 'Def_Leppard';
+      const titles = [titleString];
+      const prop = 'image';
+      const params = { ...defaultParams, prop, titles: titleString };
+      const continueParams = { tlcontinue: '46565|10|Tl', continue: '||' };
+
+      const mockedResponse1 = { data: { continue: continueParams, query: { pages: page1 } } };
+      const mockedResponse2 = { data: { batchcomplete: '', query: { pages: page1 } } };
+      axiosClient.get.mockResolvedValueOnce(mockedResponse1).mockResolvedValueOnce(mockedResponse2);
+
+      const client = new Client();
+      const subject = await client.getResultsFromApi(titles, 'image', wikiUrl);
+
+      expect(axiosClient.get).toHaveBeenCalledWith(apiUrl, { params });
+      expect(axiosClient.get).toHaveBeenCalledWith(apiUrl, {
+        params: { ...params, ...continueParams },
+      });
+      expect(subject).toEqual({ pages });
```

Comment:
Great tests 👍 
You could reduce some duplication by moving the shared `const`s one level up (like `titleString`)!